### ISSUE TITLE
Some MIR dumps differ between little and big endian

### ISSUE THREAD

**Issue by oli-obk** [rust-lang/rust#79799]
in #77413 it was noted that MIR dumps in our test suite somtimes differ between little and big endian, making our test suite fail on big endian. This is just about MIR pretty printing.

All that needs to be done is to pull out https://github.com/rust-lang/rust/blob/5ded394553296d56bb66e925d7001ab3271979ce/compiler/rustc_mir/src/util/pretty.rs#L432-L436 into a separate function that returns `true` if the verbose printing should be used and then make it recurse on `ty::Tuple` and `ty::Array` elements.

You can then run rerun tests separately for 64bit and 32bit (the latter via e.g. `--target i686-unknown-linux-gnu`)

**Comment by pierwill** [comment]
Would like to work on this! :)

@rustbot claim

**Comment by hencrice** [comment]
@rustbot claim

I would like to give this a try as my first Rust contribution :)

**Comment by infinity0** [comment]
This is still not fixed, I see the test failures still in 1.51 https://buildd.debian.org/status/fetch.php?pkg=rustc&arch=s390x&ver=1.51.0%2Bdfsg1-1%7Eexp2&stamp=1619448544&raw=0

**Comment by infinity0** [comment]
e.g. on s390x:

~~~~
-	+                                          // + literal: Const { ty: (u32, bool), val: Value(ByRef { alloc: Allocation { bytes: [2, 0, 0, 0, 0, 0, 0, 0], relocations: Relocations(SortedMap { data: [] }), init_mask: InitMask { blocks: [31], len: Size { raw: 8 } }, size: Size { raw: 8 }, align: Align { pow2: 2 }, mutability: Not, extra: () }, offset: Size { raw: 0 } }) }
+	+                                          // + literal: Const { ty: (u32, bool), val: Value(ByRef { alloc: Allocation { bytes: [0, 0, 0, 2, 0, 0, 0, 0], relocations: Relocations(SortedMap { data: [] }), init_mask: InitMask { blocks: [31], len: Size { raw: 8 } }, size: Size { raw: 8 }, align: Align { pow2: 2 }, mutability: Not, extra: () }, offset: Size { raw: 0 } }) }
~~~~

**Comment by oli-obk** [comment]
we need the same fix (as the one in #79999 ) for mir::Constant
